src/Microsoft.AspNetCore.WebSockets/WebSocketMiddleware.cs

```diff
@@ -65,19 +86,26 @@ public bool IsWebSocketRequest
             {
                 get
                 {
-                    if (!_upgradeFeature.IsUpgradableRequest)
-                    {
-                        return false;
-                    }
-                    var headers = new List<KeyValuePair<string, string>>();
-                    foreach (string headerName in HandshakeHelpers.NeededHeaders)
+                    if (_isWebSocketRequest == null)
                     {
-                        foreach (var value in _context.Request.Headers.GetCommaSeparatedValues(headerName))
+                        if (!_upgradeFeature.IsUpgradableRequest)
+                        {
+                            _isWebSocketRequest = false;
+                        }
+                        else
                         {
-                            headers.Add(new KeyValuePair<string, string>(headerName, value));
+                            var headers = new List<KeyValuePair<string, string>>();
+                            foreach (string headerName in HandshakeHelpers.NeededHeaders)
+                            {
+                                foreach (var value in _context.Request.Headers.GetCommaSeparatedValues(headerName))
```

Comment:
What happens here for headers without a value?

Reply:
Suggest [Hiding whitespace changes](https://github.com/aspnet/WebSockets/pull/252/files?utf8=%E2%9C%93&diff=split&w=1) to clarify what's actually changing in this property.